content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md

```diff
@@ -84,6 +86,53 @@ kubeadm to tell it what to do. This crashloop is expected and normal.
 After you initialize your control-plane, the kubelet runs normally.
 {{< /note >}}
 
+#### Network setup
+
+kubeadm similarly to other Kubernetes components tries to find a usable IP on
+the network interface associated with the default gateway on a host. Such
+an IP is then used for the advertising and/or listening performed by a component.
+
+To find out what this IP is on a Linux host you can use:
+
+```shell
+ip route show # Look for a line starting with "default via"
+```
+
+Kubernetes components do not accept custom network interface as an option,
+therefore a custom IP address must be passed as a flag to all components instances
+that need such a custom configuration.
+
+To configure the API server advertise address for control plane nodes created with both
+`init` and `join`, the flag `--apiserver-advertise-address` can be used.
+Preferably, this option can be set in the [kubeadm API](/docs/reference/config-api/kubeadm-config.v1beta3)
+as `InitConfiguration.localAPIEndpoint` and `JoinConfiguration.controlPlane.localAPIEndpoint`.
+
+For kubelets on all nodes, the `--node-ip` option can be passed in
+`.nodeRegistration.kubeletExtraArgs` inside a kubeadm configuration file
+(`InitConfiguration` or `JoinConfiguration`).
+
+{{< note >}}
+IP addresses become part of certificates SAN fields. Changing these IP addresses would require
+signing new certificates and restarting the affected components, so that the change in
+certificate files is reflected. See
+[Manual certificate renewal](/docs/tasks/administer-cluster/kubeadm/kubeadm-certs/#manual-certificate-renewal)
+for more details on this topic.
+{{</ note >}}
+
+{{< warning >}}
+The Kubernetes project recommends against this approach (configuring all component instances
+with custom IP addresses). Instead, the Kubernetes maintainers recommend to setup the host network,
+so that the default gateway IP is the one that Kubernetes components auto-detect and use.
+On Linux, commands such as `ip route` can be used for that purpose. If this default gateway
+IP is a public address, firewall rules must be set in place to protect the nodes and cluster.
```

Comment:
```suggestion
On Linux nodes, you can use commands such as `ip route` to configure networking; your operating
system might also provide higher level network management tools.

---

If your node's default gateway is a public IP address, you should configure packet filtering or other
security measures that protect the nodes and your cluster.
```

- Packet filtering is only one option here, even if it is the most common control.
   You could have a default gateway with a public IP address but still not need packet filtering because
   the default gateway only routes packets for private networks (that use public IP address space, 1980s
   style). You might also use have bound the control plane containers to localhost and done something
   fancy with a sidecar in the same static Pod.

Reply:
updated.